embedded/sql/type_conversion.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2022 Codenotary Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sql
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+type converterFunc func(TypedValue) (TypedValue, error)
+
+func getConverter(src, dst SQLValueType) (converterFunc, error) {
+	if src == dst {
+		return func(tv TypedValue) (TypedValue, error) {
+			return tv, nil
+		}, nil
+	}
+
+	if src == AnyType {
+		return func(val TypedValue) (TypedValue, error) {
+			if val.RawValue() == nil {
+				return &NullValue{t: dst}, nil
+			}
+			return nil, ErrInvalidValue
+		}, nil
+	}
+
+	if dst == TimestampType {
+
+		if src == IntegerType {
+			return func(val TypedValue) (TypedValue, error) {
+				if val.RawValue() == nil {
+					return &NullValue{t: TimestampType}, nil
+				}
+				return &Timestamp{val: time.Unix(val.RawValue().(int64), 0).Truncate(time.Microsecond).UTC()}, nil
+			}, nil
+		}
+
+		if src == VarcharType {
+			return func(val TypedValue) (TypedValue, error) {
+				if val.RawValue() == nil {
+					return &NullValue{t: TimestampType}, nil
+				}
+
+				str := val.RawValue().(string)
+				for _, layout := range []string{
+					"2006-01-02 15:04:05.999999",
```

Comment:
these can be kept constants? Else this array would get initialised every call

Reply:
will add constant for code readability. Not sure if it will make a difference in allocations, what do you think about it @Razikus  ;)

Reply:
in terms of performance probably it will 0-affect due to compilator optimizations